sdk/python/kfp/compiler/compiler.py

```diff
@@ -828,8 +830,18 @@ def _create_workflow(self,
     from ._data_passing_rewriter import fix_big_data_passing
     workflow = fix_big_data_passing(workflow)
 
-    import json
-    workflow.setdefault('metadata', {}).setdefault('annotations', {})['pipelines.kubeflow.org/pipeline_spec'] = json.dumps(pipeline_meta.to_dict(), sort_keys=True)
+    metadata = workflow.setdefault('metadata', {})
+    annotations = metadata.setdefault('annotations', {})
+    annotations['pipelines.kubeflow.org/sdk_version'] = 'kfp==' + kfp.__version__
```

Comment:
Perhaps we can define `pipelines.kubeflow.org/sdk_version` as a constant so that it can easily imported from other places when needed.

Also I recommend not including 'kfp==' in the version, wdyt?

Reply:
I'm not 100% sure we even need the `sdk_version` as it can be problematic in the case when multiple SDKs are used (e.g. TFX SDK + KFP SDK).
The version of KFP SDK will be provided in the `kfp_sdk_version` annotation.
The version of TFX SDK can be provided in the `tfx_sdk_version` annotation.
The goal of the `sdk_version` then will be closer to a `sdk_type`-like annotation - mostly used to specify the "main"/"root" SDK. Adding the version just makes it more informative.

Reply:
I am neutral to whether adding `sdk_version` or not. However if we do, I feel like it would be a more manageable way to let different SDK append different labels. Instead of squeezing everything into one place. So I guess I'll recommend something like `annotations['pipelines.kubeflow.org/kfp_sdk_version']` here.

In general I think it's preferred to explicitly declare things than overloading things implicitly, see [1]

[1] https://www.python.org/dev/peps/pep-0020/

Reply:
I've removed the `sdk_version` annotation.
Let's add the `tfx_sdk_version` annotation or label on the TFX side.

Reply:
SG